### PR TITLE
docs(portals): update the portal cookbook example

### PR DESCRIPTION
**Pull request by mhevery**
# Overview

# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  useStylesScoped$,
+  type ContextId,
+  type QRL,
+  type Signal,
+} from '@builder.io/qwik';
+import { type JSXNode } from '@builder.io/qwik/jsx-runtime';
+import CSS from './portal-provider.css?inline';
+
+// Define public API for opening up Portals
+export const PortalAPI = createContextId<
```
Comment (shairez) [inline packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx:18]: I think we should rename it to `PortalAPIContextId` to make it different than the actual API

File: packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+>('PortalManagerAPI');
+
+export type ContextPair<T> = { id: ContextId<T>; value: T };
+
+// Define public API for closing Portals
+export const PortalCloseAPI =
+  createContextId<QRL<() => void>>('PortalCloseAPI');
+
+// internal context for managing portals
+const Portals = createContextId<Signal<Portal[]>>('Portals');
```
Comment (shairez) [inline packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx:38]: Same comment here with the suffix of ContextId

Comment (mhevery) [reply packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx:38]: 👍

File: packages/docs/src/routes/docs/cookbook/portal/index.mdx [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+## Adding `PortalProvider` to root `layout.tsx`
+
+<CodeFile src="/src/routes/demo/cookbook/portal/layout.tsx">
+```tsx
+import { Slot, component$ } from '@builder.io/qwik';
+import { Portal, PortalProvider } from './portal-provider';
+
+export default component$(() => {
+  // 1. Wrap a root component with a <PortalManager> to enable portal API.
+  //    The <PortalManager> component will provide a context API to
```
Comment (ulic75) [inline packages/docs/src/routes/docs/cookbook/portal/index.mdx:297]: s/PortalManger/PortalProvider

File: packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+   * Add JSX to a portal.
+   * @param name portal name.
+   * @param jsx to add.
+   * @param contexts to add to the portal.
+   * @returns A function used for closing the portal.
+   */
+  QRL<
+    (name: string, jsx: JSXNode, contexts?: ContextPair<any>[]) => () => void
+  >
+>('PortalManagerAPI');
```
Comment (ulic75) [inline packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx:29]: PortalProviderAPI?

File: packages/docs/src/routes/docs/cookbook/portal/index.mdx [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+   * Add JSX to a portal.
+   * @param name portal name.
+   * @param jsx to add.
+   * @param contexts to add to the portal.
+   * @returns A function used for closing the portal.
+   */
+  QRL<
+    (name: string, jsx: JSXNode, contexts?: ContextPair<any>[]) => () => void
+  >
+>('PortalManagerAPI');
```
Comment (ulic75) [inline packages/docs/src/routes/docs/cookbook/portal/index.mdx:186]: s/PortalMangerAPI/PortalProviderAPI?

**Pull request merged**